### PR TITLE
Port Glacier2 to C++11

### PR DESCRIPTION
**Pull request by pepone**
This PR brings back to main the Glacier2 C++11 port @externl  did in https://github.com/zeroc-ice/ice/issues/629

There are some minor build fixes on top of it

### CODE REVIEW

**Review by bernardnormier — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -17,6 +17,6 @@ jobs:
       - name: Checkout repository
         uses: actions/checkout@v3
       - name: Build
-        run: make -j8 V=1
+        run: make -j8 V=1 CONFIGS="shared cpp11-shared"
```
Comment (externl) [inline .github/workflows/ci.yml:20]: Should this be `cpp98-shared cpp11-shared`?

**Review by externl — APPROVED** [review]
Looks good!

File: .github/workflows/ci.yml (hunk 1 above)
Comment (bernardnormier) [reply .github/workflows/ci.yml]: No, we call "cpp98-shared" simply "shared".

Comment (externl) [reply .github/workflows/ci.yml]: `shared` was renamed to `cpp98-shared`

**Pull request merged**